### DISCUSSION TITLE
I need to have log of each execution

### DISCUSSION THREAD

**Discussion by RandomBrown** [getsentry/sentry-python#3462] [Q&A]
Hi,

I use the logging library to log info messages for an application, and I generate a log file for each execution.

However, I want to have visibility of these messages from sentry.io, and be able to visualize them for each execution.

To achieve this, I am currently using the integration between sentry.io and logging, with a configuration of sentry_sdk.init as follows:

`integrations=[LoggingIntegration(level=logging.INFO, event_level=logging.ERROR)]`

And I manually generate an info event in sentry.io to provide observability at the end of execution, allowing me to see all the breadcrumbs:

`sentry_sdk.capture_message("[End] ### End of Execution ###", level="info")`

The result is as expected, except that sentry.io considers this level=“info” event as an error. For example, when I check the performance option, I see a 100% failure rate due to these messages.

Can I approach this from a different perspective, so that I still have this observability with each execution of the application, but without sentry.io considering it an error?

**Comment by antonpirker** [comment]
Hey @RandomBrown !

I tried this now with this example code:
```python
import os
import sentry_sdk

from sentry_sdk.integrations.logging import LoggingIntegration

import logging
logging.basicConfig(level=logging.INFO)

sentry_sdk.init(
    dsn=os.environ["SENTRY_DSN"],
    release="42.42.42",
    enable_tracing=True,
    debug=True,
    integrations=[
        LoggingIntegration(level=logging.INFO, event_level=logging.ERROR)
    ],
)

with sentry_sdk.start_transaction(op="function", name="plain-python-example-new"):
    logging.info("Hello, World! 1")
    logging.info("Hello, World! 2")
    logging.info("Hello, World! 3")
    logging.info("Hello, World! 4")
    logging.info("Hello, World! 5")
    sentry_sdk.capture_message("[End] ### End of Execution ###", level="info")
```


And I get the breadcrumbs in the message (as you described):
![Screenshot 2024-09-04 at 13 23 49](https://github.com/user-attachments/assets/cb10ecb3-c253-42cd-98f5-31a0cbc6060d)

(notice the transaction name is `plain-python-example-new`.
When I go to the performance section I do not get an error rate of 100%:
![Screenshot 2024-09-04 at 13 26 04](https://github.com/user-attachments/assets/1bde85af-381b-4ee4-ac16-4d8a89ae79d3)

So I guess this should work as you expect. Did I got something wrong?

**Reply by RandomBrown** [reply]
Hey @antonpirker !

I can't seem to replicate that behaviour. As I indicated in my reply of Sep 22, it generates failure rate increments. Surely I must be doing something wrong, any ideas?

**Comment by RandomBrown** [comment]
Hi @antonpirker :

Thanks for your comment and proposed line of work. This is the path I am using in my code. 

I set the logger level to “info”, and log events using “logger.info”.

Finally I force the saving via: sentry_sdk.capture_message("[Fin] ### Fin de Ejecucion ###", level="info")

But the result is that I get a failure rate of 100% in Performance. I attach some screenshots. 

![msedge_4M7SdedVhz](https://github.com/user-attachments/assets/55237649-2030-4472-bfb7-e4389d7bc59a)

![msedge_b7rnjObbSj](https://github.com/user-attachments/assets/1c2831bd-4f52-4163-809d-34e567180259)

![msedge_9HJ2ca4UWe](https://github.com/user-attachments/assets/9a23239f-47ee-4d73-8162-5463de7ce78e)

Any other ideas, what could I be doing wrong?

**Comment by RandomBrown** [comment]
Hi,

I have created a new project just as indicated @antonpirker , with this code:

![image](https://github.com/user-attachments/assets/eb31eab5-5c82-49ba-840c-3a52a621984d)

And I use a new project in sentry.io for this test.

But I get errors:

![image](https://github.com/user-attachments/assets/a165be2b-066a-4589-8bc5-ab870ea2d226)

![image](https://github.com/user-attachments/assets/b56e48ea-67cd-4218-85cd-6c8c4ffdca9f)

And I receive e-mail notification of the error:

![image](https://github.com/user-attachments/assets/034e7596-4c1b-4488-9b91-32ffe699e91b)

How can I solve this? 

Thanks!!!

**Comment by RandomBrown** [comment]
Finally, I managed to fix it. I’ll explain how I did it in case it’s of interest to anyone. The mistake I was making was logging informational messages using “capture_message,” which always creates an event in Sentry. These events were affecting the performance metrics.

I replaced it with transactions using “start_transaction” instead of “capture_message.”

My problem now is that for a specific execution on a certain date and time that didn’t log any errors, I want to see the information I recorded, using the integration between sentry.io and logging. How can I view it?
